flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -237,8 +254,15 @@ public boolean isEmpty() {
 						computationState.getCounter());
 
 				newComputationStates = Collections.emptyList();
+				nfaChanged = true;
```

Comment:
Just a question/side comment. This case always happens in pair with pruning sharedBuffer right? (NFA:330)

I would still leave it here, cause it's better be safe than sorry.